routers/api/v1/repo/tree.go

```diff
@@ -110,14 +111,19 @@ func GetTreeBySHA(ctx *context.APIContext, sha string) *gitea.GitTreeResponse {
 	if page <= 0 {
 		page = 1
 	}
+	tree.Page = page
+	tree.TotalCount = len(entries)
+	tree.TotalPages = int(math.Ceil(float64(len(entries)) / float64(perPage)))
```

Comment:
I think this should be calculated at caller side. We could replace `PageSize` probably instead